src/python/pants/backend/experimental/codegen/protobuf/scala/register.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from pants.backend.codegen.protobuf import protobuf_dependency_inference
+from pants.backend.codegen.protobuf import target_types as protobuf_target_types
+from pants.backend.codegen.protobuf.scala import rules as scala_protobuf_rules
+from pants.backend.codegen.protobuf.target_types import (
+    ProtobufSourcesGeneratorTarget,
+    ProtobufSourceTarget,
+)
+from pants.backend.experimental.scala.register import rules as all_scala_rules
+
+
+def target_types():
+    return [ProtobufSourcesGeneratorTarget, ProtobufSourceTarget]
+
+
+def rules():
+    return [
+        *all_scala_rules(),
```

Comment:
This shouldn't be necessary... does the backend loading integration test fail without it? The effect of importing all of these would be that enabling only this backend would add ~all of the scala backend, without defining any of its targets. While it wouldn't really make sense to do that, it would be a bit better to be minimal here.

Reply:
It does:

https://github.com/pantsbuild/pants/runs/4487021732?check_suite_focus=true#step:13:563

```
 E       AssertionError: Failed to load: ['pants.backend.experimental.codegen.protobuf.scala']
E         /home/runner/.cache/pants/named_caches/pex_root/venvs/s/c73ea1e4/venv/bin/python3.7 -m pants --no-pantsrc --pants-workdir=/tmp/process-executionmyXdzT/.pants.d/tmp/tmpe6fgatgf.pants.d --print-stacktrace=True --pantsd --pants-config-files=[] --pants-config-files=/tmp/process-executionmyXdzT/.pants.d/tmp/tmpe6fgatgf.pants.d/pants.toml --no-verify-config --version
E         exit_code: 1
E         stdout:
E         	
```

Reply:
Since your unit tests don't import "the whole backend", I expect that re-exporting whatever was necessary to get the tests passing (probably from the codegen backend itself, so that you don't have to do it in two places) would be sufficient. But not a blocker.